src/qtism/runtime/storage/binary/QtiBinaryStreamAccess.php

```diff
@@ -78,22 +80,30 @@ class QtiBinaryStreamAccess extends BinaryStreamAccess
 
     const RW_CORRECTRESPONSE = 2;
 
-    /**
-     * @var FileManager
-     */
+    private const VARIABLE_TYPES = [
+        'outcomeDeclaration',
+        'responseDeclaration',
+        'templateDeclaration',
+    ];
```

Comment:
Please add indexes. The integer values encoded in the session have fixed values (0 = outcomeVariable, 1 = responseVariable, 2 = templateVariable) and not having them in the mapping obfuscate the fact that they matter. If you don't want to rely on magic number, constants as keys are even better.